context/app/static/js/pages/Diversity/Diversity.jsx

```diff
@@ -14,164 +14,44 @@ import ProjectAttribution from './ProjectAttribution';
-             "filter": {
-                 "term": {
-                     "entity_type": "donor"
-                 }
-             }
-         }
-     },
-     "aggs": {
-         "mapped_metadata.sex": {
-             "terms": {
-                 "field": "mapped_metadata.sex.keyword"
-             }
-         }
-     }
- }
-
- */
-const donorRaceSexQuery = {
-  size: 0,
-  aggs: {
-    composite_data: {
-      composite: {
-        sources: [
-          {
-            'mapped_metadata.race': {
-              terms: {
-                field: 'mapped_metadata.race.keyword',
-              },
-            },
-          },
-          {
-            'mapped_metadata.blood_type': {
-              terms: {
-                field: 'mapped_metadata.blood_type.keyword',
-              },
-            },
-          },
-        ],
-        size: 10000,
-      },
+function termSource(field) {
+  const esField = `mapped_metadata.${field}`;
+  const source = {};
+  source[esField] = {
+    terms: {
+      field: `mapped_metadata.${field}.keyword`,
     },
-  },
-};
+  };
+  return source;
+}
 
-const donorGenderRace = {
-  size: 0,
-  aggs: {
-    composite_data: {
-      composite: {
-        sources: [
-          {
-            'mapped_metadata.race': {
-              terms: {
-                field: 'mapped_metadata.race.keyword',
-              },
-            },
-          },
-          {
-            'mapped_metadata.sex': {
-              terms: {
-                field: 'mapped_metadata.sex.keyword',
-              },
-            },
-          },
-        ],
-        size: 10000,
-      },
+function histogramSource(field) {
+  const esField = `mapped_metadata.${field}`;
+  const source = {};
+  source[esField] = {
+    histogram: {
+      field: `${esField}_value`,
+      interval: 10,
     },
-  },
-};
-
-const donorBloodtypeGender = {
-  size: 0,
-  aggs: {
-    composite_data: {
-      composite: {
-        sources: [
-          {
-            'mapped_metadata.sex': {
-              terms: {
-                field: 'mapped_metadata.sex.keyword',
-              },
-            },
-          },
-          {
-            'mapped_metadata.blood_type': {
-              terms: {
-                field: 'mapped_metadata.blood_type.keyword',
-              },
-            },
-          },
-        ],
-        size: 10000,
-      },
-    },
-  },
-};
+  };
+  return source;
+}
 
-const donorAgeGender = {
-  size: 0,
-  aggs: {
-    composite_data: {
-      composite: {
-        sources: [
-          {
-            'mapped_metadata.age': {
-              histogram: {
-                field: 'mapped_metadata.age_value',
-                interval: 10,
-              },
-            },
-          },
-          {
-            'mapped_metadata.sex': {
-              terms: {
-                field: 'mapped_metadata.sex.keyword',
-              },
-            },
-          },
-        ],
-        size: 10000,
+function compositeQuery(source1, source2) {
```

Comment:
Some of these could be useful outside of this component, consider adding them to shared utils somewhere.